File: Sources/windows/BitTherapyCLR/src/sprites/sprite_set.cpp
#include "sprite.h"
#include "sprite_set.h"

#include <iostream>
#include <string>
#include <vector>

SpriteSet::SpriteSet(
    const std::map<std::string, std::vector<std::string>> animations
) : animations(animations) {}

SpriteSet::SpriteSet() : 
    animations(std::map<std::string, std::vector<std::string>>({})) {}

Sprite SpriteSet::sprite(const std::string animationName, double fps) const {
    auto frames = spriteFrames(animationName);
    Sprite sprite(animationName, frames, fps);
    return sprite;
}

const std::vector<std::string> SpriteSet::spriteFrames(const std::string animationName) const {
    if (auto iter = animations.find(animationName); iter != animations.end()) {
       return iter->second;
    }
    return {""};
}

File: Sources/windows/App/Sources/Sprites/SpriteSet.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace App
{
    public class SpriteSet
    {
        private Dictionary<string, List<string>> animations;

        public SpriteSet(Dictionary<string, List<string>> animations)
        {
            this.animations = animations;
        }

        public SpriteSet() : this(new Dictionary<string, List<string>>())
        {
        }

        public List<string> SpriteFrames(string animationName)
        {
            return animations.TryGetValue(animationName, out var frames) ? frames : new List<string> { "" };
        }

        public Sprite CreateSprite(string animationName, double fps)
        {
            var frames = SpriteFrames(animationName);
            return new Sprite(animationName, frames, fps);
        }
    }
}
